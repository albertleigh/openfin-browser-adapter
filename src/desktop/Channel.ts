import {BaseApiClass} from './Base';
import {ChannelProvider} from './ChannelProvider';
import {ChannelClient} from './ChannelClient';
import {OpenFinMock} from '../OpenFinMock'

const _ChannelProvider = OpenFinMock.generateMethods('InterApplicationBus.Channel#ChannelProvider',ChannelProvider);
const _ChannelClient = OpenFinMock.generateMethods('InterApplicationBus.Channel#ChannelClient',ChannelClient);

export class Channel extends BaseApiClass{

    static staticMethods:string[]=[
        'onChannelConnect',
        'onChannelDisconnect',
    ];

    static instanceMethods:string[]=[];

    static async create(){
        return new _ChannelProvider();
    }

    static async connect(){
        return new _ChannelClient();
    }

}