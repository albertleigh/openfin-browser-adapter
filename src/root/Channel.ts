import {BaseApiClass} from './Base';
import {ChannelProvider} from './ChannelProvider';
import {ChannelClient} from './ChannelClient';
import {OpenFinMockV2} from '../OpenFinMockV2'

const _ChannelProvider = OpenFinMockV2.generateMethods('InterApplicationBus.Channel#ChannelProvider',ChannelProvider);
const _ChannelClient = OpenFinMockV2.generateMethods('InterApplicationBus.Channel#ChannelClient',ChannelClient);

export class Channel extends BaseApiClass{

    static staticMethods:string[]=[
        'onChannelConnect',
        'onChannelDisconnect',
    ];

    static instanceMethods:string[]=[];

    static async create(){
        return new _ChannelProvider();
    }

    static async connect(){
        return new _ChannelClient();
    }

}