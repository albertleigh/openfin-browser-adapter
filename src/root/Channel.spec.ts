import { Channel } from './Channel';

describe('Desktop::Channel',()=>{

    let clazz:typeof Channel

    beforeEach(()=>{
        clazz = Channel;
    })

    it('all unsupported static methods defined',()=>{
        expect(clazz.staticMethods).toMatchSnapshot();
    })

    it('all unsupported instance methods defined',()=>{
        expect(clazz.instanceMethods).toMatchSnapshot();
    })

    it('create',async ()=>{
        expect(await clazz.create()).toMatchSnapshot();
    })

    it('connect',async ()=>{
        expect(await clazz.connect()).toMatchSnapshot();
    })

})