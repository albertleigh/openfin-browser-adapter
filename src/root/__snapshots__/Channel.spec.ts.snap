// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop::Channel all unsupported instance methods defined 1`] = `Array []`;

exports[`Desktop::Channel all unsupported static methods defined 1`] = `
Array [
  "onChannelConnect",
  "onChannelDisconnect",
]
`;

exports[`Desktop::Channel connect 1`] = `ChannelClient {}`;

exports[`Desktop::Channel create 1`] = `ChannelProvider {}`;
